Lesson_1/HelloName_Lesson/HelloName_Lesson/AutoParkClasses/Transport.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+namespace HelloName_Lesson.AutoParkClasses
+{
+    abstract public class Transport : IComparable
+    {
+        public string destination { get; set; }
+        public int transportNumber { get; set; }
+        public string departureTime { get; set; }
+        public int amountOfSeats { get; set; }
+
+        public Transport() { }
+
+        public virtual void GetInfo() { }
+
+        abstract public string GetTheTransportType();
+
+        public virtual int CompareTo(object? obj)
+        {
+            if (obj is Transport transport)
+            {
+                return amountOfSeats.CompareTo(transport.amountOfSeats);
+            }
+            return 0;
+        }
+    }
```

Comment:
We khow generic and you can use https://learn.microsoft.com/pl-pl/dotnet/api/system.icomparable-1?view=net-7.0